src/utils/helpers.ts

```diff
@@ -94,3 +91,81 @@ export const getERC20Record = (
 
   return null;
 };
+
+export const extractTextContent = (msg: ChatMessage): string => {
+  if (typeof msg.content === 'string') {
+    return msg.content;
+  }
+
+  if (Array.isArray(msg.content)) {
+    return msg.content
+      .filter((item) => item.type === 'text')
+      .map((item) => item.text)
+      .join(' ');
+  }
+
+  return '';
+};
```

Comment:
Ensures that we are only searching through text content and not file upload urls, for instance